File: ueueo-framework/ueueo-distributed-locking-abstractions/src/main/java/Abp/DistributedLocking/LocalAbpDistributedLock.java
package Abp.DistributedLocking;

import com.ueueo.threading.CancellationToken;

import java.util.Objects;
import java.util.concurrent.TimeUnit;

/**
 * TODO Description Of This JAVA Class.
 *
 * @author Lee
 * @date 2022-05-29 15:46
 */
public class LocalAbpDistributedLock implements IAbpDistributedLock {

    @Override
    public IAbpDistributedLockHandle tryAcquire(String name, Integer timeout, TimeUnit timeUnit, CancellationToken cancellationToken) {
        Objects.requireNonNull(name);

        return null;
    }

}

File: ueueo-framework/ueueo-distributed-locking-abstractions/src/main/java/Abp/DistributedLocking/LocalAbpDistributedLock.cs
using System;
using System.Collections.Concurrent;
using System.Threading;
using System.Threading.Tasks;
using Volo.Abp.DependencyInjection;

namespace Volo.Abp.DistributedLocking;

public class LocalAbpDistributedLock : IAbpDistributedLock, ISingletonDependency
{
    private readonly ConcurrentDictionary<String, SemaphoreSlim> _localSyncObjects = new();

    public  Task<IAbpDistributedLockHandle> TryAcquireAsync(
        String name,
        TimeSpan timeout = default,
        CancellationToken cancellationToken = default)
    {
        Check.NotNullOrWhiteSpace(name, nameof(name));

        var semaphore = _localSyncObjects.GetOrAdd(name, _ => new SemaphoreSlim(1, 1));

        if (!semaphore.WaitAsync(timeout, cancellationToken))
        {
            return null;
        }

        return new LocalAbpDistributedLockHandle(semaphore);
    }
}
